server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -256,7 +256,13 @@ public Form<EditTiClientInfoForm> updateClientInfo(
   public TrustedIntermediarySearchResult getManagedAccounts(
       SearchParameters searchParameters, TrustedIntermediaryGroupModel tiGroup) {
     ImmutableList<AccountModel> allAccounts = tiGroup.getManagedAccounts();
-    if (searchParameters.nameQuery().isEmpty() && searchParameters.dateQuery().isEmpty()) {
+    /* Filtering to transform empty strings into empty optionals.
+    Empty parameters should return all accounts. */
+    Optional<String> filteredNameQuery =
+        searchParameters.nameQuery().filter(s -> !s.isEmpty() && !s.isBlank());
```

Comment:
are we handling null cases?

Reply:
Since the `nameQuery` and `dateQuery` fields are of type `Optional`, they're initialized as empty `Optional`s rather than `null`.  Also, in practice, we set them to empty strings as a default here:  https://github.com/civiform/civiform/blob/40e4fa72e40a2ef0fb7f7c1117753bb9d23d6086/server/app/views/applicant/TrustedIntermediaryDashboardView.java#L118.